### PR TITLE
[IMP] product: add store and index on name of attributes and values

### PR DESCRIPTION
**Pull request by seb-odoo**
For performance since those might be searched, or displayed without the related
model necessarily in cache.

Description of the issue/feature this PR addresses:

Current behavior before PR:

Desired behavior after PR is merged:

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Comment by JKE-be** [comment]
@nseinlet 

can I have your opinion for this one plz

**Comment by JKE-be** [comment]
@robodoo r+

**Comment by nseinlet** [comment]
@robodoo r-

File: addons/product/models/product_attribute.py [hunk 1]
```diff
@@ -221,7 +221,7 @@ class ProductTemplateAttributeValue(models.Model):
     _order = 'product_attribute_value_id, id'
     _description = 'Product Attribute Value'
 
-    name = fields.Char('Value', related="product_attribute_value_id.name")
+    name = fields.Char('Value', related="product_attribute_value_id.name", store=True, index=True)
```
Comment (nseinlet) [inline addons/product/models/product_attribute.py]: storing a related translatable field doesn't seems efficient.

File: addons/product/models/product_attribute.py [hunk 2]
```diff
@@ -117,10 +117,10 @@ class ProductTemplateAttributeLine(models.Model):
     Used as a configuration model to generate the appropriate product.template.attribute.value"""
 
     _name = "product.template.attribute.line"
-    _rec_name = 'attribute_id'
     _description = 'Product Template Attribute Line'
     _order = 'attribute_id, id'
 
+    name = fields.Char('Value', related='attribute_id.name', store=True, index=True)
```
Comment (nseinlet) [inline addons/product/models/product_attribute.py]: storing a related translatable field doesn't seems efficient.

File: addons/product/models/product_attribute.py [hunk 3]
```diff
@@ -66,7 +66,7 @@ class ProductAttributeValue(models.Model):
     _order = 'attribute_id, sequence, id'
     _description = 'Attribute Value'
 
-    name = fields.Char(string='Value', required=True, translate=True)
+    name = fields.Char(string='Value', required=True, translate=True, index=True)
```
Comment (nseinlet) [inline addons/product/models/product_attribute.py]: As the field is translatable, is the index really efficient?
As the index is a BTree index, do you use startswith or equal comparison?

File: addons/product/models/product_attribute.py [hunk 4]
```diff
@@ -14,7 +14,7 @@ class ProductAttribute(models.Model):
     # `_sort_key_attribute_value` in `product.template`
     _order = 'sequence, id'
 
-    name = fields.Char('Attribute', required=True, translate=True)
+    name = fields.Char('Attribute', required=True, translate=True, index=True)
```
Comment (nseinlet) [inline addons/product/models/product_attribute.py]: As the field is translatable, is the index really efficient?
As the index is a BTree index, do you use startswith or equal comparison?

**Pull request closed without merging**